src/action.js

```diff
@@ -255,19 +280,61 @@ async function setStatus(pullRequest, state, description) {
   });
 }
 
+async function generateReleaseNotes(pullRequest) {
+  const tag = getTagName(pullRequest);
+
+  var previous_tag_name;
+  try {
+    const latest_release = await client.rest.repos.getLatestRelease({
+      owner: github.context.repo.owner,
+      repo: github.context.repo.repo,
+    });
+    previous_tag_name = latest_release.data.tag_name;
+    core.info(
+      "Generating release-notes relative to release " + previous_tag_name + ".."
+    );
+  } catch (error) {
+    previous_tag_name = "";
+    core.info("Generating release-notes relative to start of repository..");
+  }
+
+  try {
+    const response = await client.request(
+      "POST /repos/{owner}/{repo}/releases/generate-notes",
+      {
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        tag_name: tag,
+        target_commitish: pullRequest.head.ref,
+        previous_tag_name: previous_tag_name,
+      }
+    );
+
+    pullRequest.body = response.data.body;
```

Comment:
Because I now mutate the actual body, I'm keeping this line in to prevent stale data in the pullRequest variable. If I don't, validateBody will tell you the description is missing when it isn't.

Reply:
That's how it used to work, but after this last revision, the rest of the code is completely unaware that the description wasn't written by people. If you think not mutating the object and instead having `validateBody` check, that's an easy fix, so I can go ahead and do that, but I don't know if I agree that it's the best solution. Under normal circumstances I agree that mutating the object is bad practice, but considering we just mutated the actual PR, I don't see it being an issue. You make the call, I don't know what is better for this code at the end of the day.

Reply:
I think we can be more explicit by adding a `releaseNotes` argument to the validate function, something like this (I have not tested this):
```diff
diff --git a/src/action.js b/src/action.js
index 73ae4e7..d454235 100644
--- a/src/action.js
+++ b/src/action.js
@@ -19,15 +19,19 @@ async function action() {
     core.info(`Validating pull request for action ${action}.`);
 
     // If the release notes should have changed, generate new ones
+    let releaseNotes;
     if (
       JSON.parse(core.getInput("generate_release_notes") || false) === true && //Should we generate notes?
       (action === "opened" || //Was the action "opened" or
         (action === "edited" && //Was it "edited" and
           github.context.payload.changes.body.from == pullRequest.body)) //The edit wasn't of the description
-    )
-      await generateReleaseNotes(pullRequest);
+    ) {
+      releaseNotes = await generateReleaseNotes(pullRequest);
+    } else {
+      releaseNotes = pullRequest.body;
+    }
 
-    await validate(pullRequest);
+    await validate(pullRequest, releaseNotes);
   } else if (action === "closed" && pullRequest.merged) {
     // Previously validated PR was merged so create release
     core.info(`Creating release.`);
@@ -42,7 +46,7 @@ async function action() {
   core.setOutput("release", pullRequest.title);
 }
 
-async function validate(pullRequest) {
+async function validate(pullRequest, releaseNotes) {
   const doValidate = JSON.parse(core.getInput("validate") || true) === true;
 
   if (!doValidate) {
@@ -55,7 +59,7 @@ async function validate(pullRequest) {
     // Check title
     validateTitle(pullRequest);
     // Check body
-    validateBody(pullRequest);
+    validateReleaseNotes(releaseNotes);
     // Check that PR is from base to head
     validateBranches(pullRequest);
     // Check that release doesn't already exist
@@ -123,10 +127,9 @@ function validateTitle(pullRequest) {
   }
 }
 
-function validateBody(pullRequest) {
-  core.info("Validating body..");
-  const { body } = pullRequest;
-  if (!body) {
+function validateReleaseNotes(releaseNotes) {
+  core.info("Validating releaseNotes..");
+  if (!releaseNotes) {
     throw new ValidationError("Missing description.");
   }
 }
@@ -301,9 +304,7 @@ async function generateReleaseNotes(pullRequest) {
       issue_number: pullRequest.number,
       body: response.data.body,
     });
-
-    //pullRequest now holds stale data, updating it
-    pullRequest.body = response.data.body;
+    return response.data.body;
   } catch (error) {
     throw Error("Failed to generate a body: " + error);
   }
@@ -340,7 +341,7 @@ module.exports = {
   action,
   validate,
   validateTitle,
-  validateBody,
+  validateReleaseNotes,
   validateBranches,
   validateRelease,
   getTagName,
```